aopy/data/db.py

```diff
@@ -414,6 +443,96 @@ def get_db_object(self):
         '''
         return self.record
 
+class BMI3DDecoder():
+    '''
+    Wrapper for BMI3D Decoder objects
+    '''
+    
+    def __init__(self, decoder, dbname='default'):
+        '''
+        Constructor
+
+        Args:
+            task_entry (TaskEntry): a database task entry from BMI3D
+            dbname (str, optional): name of the database to connect to. Defaults to 'default'.
+        '''
+        self.dbname = dbname
+        self.record = decoder
+        self.id = self.record.id
+        self.parent = lookup_sessions(id=decoder.entry_id)[0]
```

Comment:
Can decoder.entry_id be empty or None?

Reply:
good question, i don't think so but it can't hurt to check